Marlin/Conditionals_post.h

```diff
@@ -730,11 +730,16 @@
   /**
    * Set granular options based on the specific type of leveling
    */
+
+  #if ENABLED(AUTO_BED_LEVELING_UBL) && ENABLED(DELTA)
+    #define UBL_DELTA
```

Comment:
I tried very hard to make sure that my changes would not affect existing AUTO_BED_LEVELING_UBL on CARTESIAN, hence a separate UBL_DELTA "feature flag" for clarity.

UBL_DELTA is probably a misnomer.  UBL_GRANULAR_SEGMENTATION_WITH_KINEMATIC_SUPPORT is probably more accurate but a little unwieldy.

I also copied the kinematic code for SCARA, but SCARA won't build with AUTO_BED_LEVELING_UBL because there are no NEWPANEL LCD controllers defined in default SCARA configuration.h.  Otherwise UBL_DELTA should work for SCARA.

Also, most of my changes can ALSO be used on CARTESIAN if UBL_DELTA is also defined on CARTESIAN, using the new segmentation routine instead of existing ubl_line_to_destiation that segments only on mesh cell boundaries.  So again, I needed a distinct feature flag, not simply ENABLED(AUTO_BED_LEVELING_UBL) && ENABLED(DELTA).